src/System.IO.FileSystem.DriveInfo/src/System.IO.FileSystem.DriveInfo.csproj

```diff
@@ -19,6 +19,12 @@
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU' " />
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU' " />
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU' "/>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU' "/>
+  <PropertyGroup>
+    <TargetsBSD Condition="'$(TargetsOSX)' == 'true' or '$(TargetsFreeBSD)' == 'true'">true</TargetsBSD>
+  </PropertyGroup>
```

Comment:
Starting another "is BSD" condition into the code makes it very very complicated, since it means we had two layers of generic Unix-flavours. Written in a set notation: 

BSD = {FreeBSD, OS X} (I still very much deny OS X being a BSD), 
Unix = {Linux, BSD} = {Linux, {FreeBSD, OS X}}.

As I said we discussed this topic in the FreeBSD porting team and feel it is a path that will cause more problems than it solves in the future.

We should stick to the solution we used before. If it is common among most Unix platforms, it should be implemented in a common "Unix" file. If one platform, e.g. Linux, does not have it, add a file just for LInux specializing the behaviour away from the "normal" Unix implementation.

Also OS X is not really a BSD. It is a strange hybrid, with some parts coming from FreeBSD while other parts are completely different.